src/receive_imf.rs

```diff
@@ -1269,7 +1269,7 @@ RETURNING id
 
     if let Some(replace_msg_id) = replace_msg_id {
         // "Replace" placeholder with a message that has no parts.
-        replace_msg_id.delete_from_db(context).await?;
+        replace_msg_id.trash(context).await?;
```

Comment:
this changes logic in used code from "delete record physically from database" to "mark record for deletion" - i was stumbling upon that first as a replacement would not work if the id is still in the database.

however, looking closer, the replacements are done before and this line is only executed if there is _no_ replacement (this i also what the comment says)

so, all in all, that should be fine as well.

@link2xt what is the reason for this change, however? to avoid re-downloading of things as the comment in `trash()` says?